pkg/ring/kv/dynamodb/client.go

```diff
@@ -185,6 +195,17 @@ func (c *Client) CAS(ctx context.Context, key string, f func(in interface{}) (ou
 			return c.kv.Batch(ctx, putRequests, deleteRequests)
 		}
 
+		if len(putRequests) == 0 && len(deleteRequests) == 0 {
+			// no change detected, wait for 1 sec and retry
+			select {
+			case <-time.After(noChangeDetectedRetrySleep):
```

Comment:
Do we need this? I think we have the retry wait and context check in `bo.Ongoing` already. The `backoff` should already take care of this

Reply:
got it. I'll remove this wait then. The back indeed wait at least 1 sec.